src/radical/entk/task.py

```diff
@@ -1,25 +1,110 @@
-warnings.simplefilter(action="once", category=DeprecationWarning, lineno=786)
-warnings.simplefilter(action="once", category=DeprecationWarning, lineno=954)
+
+# ------------------------------------------------------------------------------
+#
+class ReqsMixin:
+
+    # --------------------------------------------------------------------------
+    #
+    def _correct_deprecated_key(self, k):
+        if k in self._deprecated_keys:
+            warnings.warn('Key "%s" is obsolete, new key ' % k +
+                          '"%s" will be used' % self._deprecated_keys[k],
+                          DeprecationWarning, stacklevel=2)
+            k = self._deprecated_keys[k]
+        return k
+
+    # --------------------------------------------------------------------------
+    #
+    def __setitem__(self, k, v):
+        k = self._correct_deprecated_key(k)
+        super().__setitem__(k, v)
+
+    # --------------------------------------------------------------------------
+    #
+    def __setattr__(self, k, v):
+        k = self._correct_deprecated_key(k)
+        super().__setattr__(k, v)
+
+
+# ------------------------------------------------------------------------------
+#
+class CpuReqs(ReqsMixin, ru.Munch):
+
+    _check = True
+
+    _schema = {
+        'cpu_processes'   : int,
+        'cpu_process_type': str,
+        'cpu_threads'     : int,
+        'cpu_thread_type' : str
+    }
+
+    _defaults = {
+        'cpu_processes'   : 1,
+        'cpu_process_type': None,
+        'cpu_threads'     : 1,
+        'cpu_thread_type' : None
+    }
+
+    # deprecated keys will cause a deprecation warning (ReqsMixin), and
+    # correct keys, according to CpuReqs schema, will be provided
+    _deprecated_keys = {
+        'processes'          : 'cpu_processes',
+        'process_type'       : 'cpu_process_type',
+        'threads_per_process': 'cpu_threads',
+        'thread_type'        : 'cpu_thread_type'
+    }
+
+
+# ------------------------------------------------------------------------------
+#
+class GpuReqs(ReqsMixin, ru.Munch):
+
+    _check = True
+
+    _schema = {
+        'gpu_processes'   : int,
+        'gpu_process_type': str,
+        'gpu_threads'     : int,
+        'gpu_thread_type' : str
+    }
+
+    _defaults = {
+        'gpu_processes'   : 0,
+        'gpu_process_type': None,
+        'gpu_threads'     : 1,
+        'gpu_thread_type' : None
+    }
+
+    # deprecated keys will cause a deprecation warning (ReqsMixin), and
+    # correct keys, according to GpuReqs schema, will be provided
+    _deprecated_keys = {
+        'processes'          : 'gpu_processes',
+        'process_type'       : 'gpu_process_type',
+        'threads_per_process': 'gpu_threads',
+        'thread_type'        : 'gpu_thread_type'
+    }
 
 
 # ------------------------------------------------------------------------------
 #
-class Task(object):
-    '''
+class Task(ru.Munch):
+    """
```

Comment:
nitpick: we actually use `'''` throughout most parts of the code.

Reply:
actually following docstring convention it is recommended to use triple double quotes ([PEP257](https://www.python.org/dev/peps/pep-0257/) referred from [PEP8](https://www.python.org/dev/peps/pep-0008/#string-quotes))

> we actually use `'''` throughout most parts of the code.

not within EnTK ;) (almost all modules here use triple double quotes)